internal/v1/handler.go

```diff
@@ -27,6 +27,21 @@ const (
 	FSMaxSize = 68719476736
 )
 
+func NewComposeResponseItemFromEntry(c db.ComposeEntry) (ComposesResponseItem, error) {
+	var cmpr ComposeRequest
+	err := json.Unmarshal(c.Request, &cmpr)
+	if err != nil {
+		return ComposesResponseItem{}, err
+	}
+	return ComposesResponseItem{
+		CreatedAt: c.CreatedAt.Format(time.RFC3339),
+		Id:        c.Id,
+		ImageName: c.ImageName,
+		Request:   cmpr,
+		ClientId:  (*ClientId)(c.ClientId),
+	}, nil
+}
+
```

Comment:
Don’t like it here either and I know we are doing this in our codebase in the payload package but that is a different use case: the function is exported (like here) and used in a different package (service) which is not the case here.

Unless you plan to use it in a different package, I suggest to stick with a struct literal.

Reply:
I do not insist on anything here, just couple of ideas here and there.